File: feign-clients/src/main/java/com/otbs/feign/dto/EmployeeResponse.java
package com.otbs.feign.dto;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

public record EmployeeResponse(
        @JsonDeserialize(as = String.class) String id,
        String username,
        String firstName,
        String lastName,
        String email,
        String department,
        String role,
        String jobTitle
) {

}

File: app/src/main/java/com/example/onetechbs/db/EmployeeResponse.kt
package com.example.onetechbs.db

data class EmployeeResponse(
    val id: String,
    val username: String,
    val role: String,
    val firstName: String,
    val lastName: String,
    val email: String,
    val jobTitle: String
    // Add other fields from your backend as neede
)